src/test/java/io/wgj/ctci/ArrayListTest.java

```diff
@@ -0,0 +1,136 @@
+package io.wgj.ctci;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.util.Iterator;
+
+import static org.junit.Assert.*;
+
+public class ArrayListTest {
+  @Rule
+  public final ExpectedException exception = ExpectedException.none();
+
+  @Test
+  public void arrayList_new() {
+    ArrayList<Integer> al = new ArrayList<>();
+  }
+
+  @Test
+  public void arrayList_add() {
+    ArrayList<Integer> al = new ArrayList<>();
+    al.add(1);
+  }
+
+  @Test
+  public void arrayList_size() {
+    ArrayList<Integer> al = new ArrayList<>();
+    assertEquals(0, al.size());
+    al.add(1);
+    assertEquals(1, al.size());
+  }
+
+  @Test
+  public void arrayList_grow() {
+    ArrayList<Integer> al = new ArrayList<>();
```

Comment:
Nit: not everyone would agree with this but for the sake of communicating intent in code I'd avoid abbreviations when space permits it because they are often ambiguous.  In this case I think "list" would be a good replacement.

Reply:
Thanks! `list` is a good compromise and falls on the right side of terse.